Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
Brett is exactly right. Connections can are re-used once the limit of active connections is reached. The oldest active connection at that point will be recycled. 

Sharing connections between threads has the potential to include queries from unrelated requests into transactions from other requests unexpectedly.